Lib/_strptime.py

```diff
@@ -567,8 +567,31 @@ def _strptime_time(data_string, format="%a %b %d %H:%M:%S %Y"):
     tt = _strptime(data_string, format)[0]
     return time.struct_time(tt[:time._STRUCT_TM_ITEMS])
 
-def _strptime_datetime(cls, data_string, format="%a %b %d %H:%M:%S %Y"):
-    """Return a class cls instance based on the input string and the
+def _strptime_datetime_date(cls, data_string, format="%a %b %d %Y"):
+    """Return a date instance based on the input string and the
+    format string."""
+    tt, _, _ = _strptime(data_string, format)
```

Comment:
Should we check the format here? The previous pull request used a C function for that, but that obviously wouldn't work if the C implementation isn't available.

Reply:
I think we do need to check that this doesn't have any non-date components in it, no? We might need to push that into `_strptime`, though. We could check that it returns all 0s, but then things like `2024-01-01T00:00` will fail to raise an exception.

Reply:
Shouldn't `strptime()` be able to parse the output produced by `strftime()`? I would prefer if both raised an error for the format or neither, but checking for the fallback date or time seems fine to me.

```python
>>> datetime.date(9999, 12, 31).strftime("%Y-%m-%dT%H:%M:%S")
'9999-12-31T00:00:00'
>>> datetime.time(23, 59, 59).strftime("%Y-%m-%dT%H:%M:%S")
'1900-01-01T23:59:59'
```

Reply:
Hmm... Good point. I guess we can just leave it without doing a check.